peekaboo/daemon.py

```diff
@@ -199,7 +246,14 @@ def run():
         pidfile.write("%s\n" % pid)
 
     systemd = SystemdNotifier()
-    server = PeekabooStreamServer(config.sock_file, PeekabooStreamRequestHandler)
+    # Try three times to start SocketServer
+    for i in range(0, 3):
+        try:
+            server = PeekabooStreamServer(config.sock_file, PeekabooStreamRequestHandler)
+            break
+        except socket.error, msg:
+            logger.warning("SocketServer couldn't start (%i)" % i)
+
     runner = Thread(target=server.serve_forever)
```

Comment:
If all three attempts to start the server fail, the program crashes, because ``server`` is undefined. You should check if ``server`` is initialized, before creating a thread that starts the server.